nab/detectors/goodness_of_fit/goodness_of_fit_detector.py

```diff
@@ -0,0 +1,156 @@
+import numpy
+import math
+from scipy import stats
+
+from nab.detectors.base import AnomalyDetector
+
+
+
+class GoodnessOfFitDetector(AnomalyDetector):
+
+  """ This detector is an implementation of online anomaly detection using
+  Relative Entropy statistic with multiple hypothesis as described in
+  Figure 1 of Wang, Chengwei, et al. "Statistical Techniques for Online
+  Anomaly Detection in Data Centers", Proceedings of the 8th ACM
+  international conference on Autonomic computing. ACM, 2011.
+
+  The algorithm is based on a hypothesis testing approach that compares
+  observed data against multiple null hypotheses, representing frequencies of
+  quantized data over a window. If the observed data is unseen and does not
+  agree with any existing hypothesis, it is declared anomolous and a new
+  hypothesis is created. Otherwise, it is declared non-anomolous, provided that
+  the accepted hypothesis occurs frequently enough. Decision to accept/reject a
+  null hypothesis is based on relative entropy compared against a threshold
+  of acceptable false negative probability determined by the chi-squared
+  distribution. Step-by-step details are given in code comments and parameters
+  have been tuned for best performance of NAB.
+  """
+
+  def __init__(self, *args, **kwargs):
+    super(GoodnessOfFitDetector, self).__init__(*args, **kwargs)
```

Comment:
I'm okay with keeping the variable names as-is to match the paper, but there should be a docstring here that says that.